ROSCO_toolbox/linear/robust_scheduling.py

```diff
@@ -0,0 +1,413 @@
+                self.om_doe.set_val('r_sched.u_eval', u)
+                self.doe_logfile = os.path.join(
+                    self.output_dir, self.output_name + '.' + str(u) + ".doe.sql")
+                self.om_doe = self.setup_recorder(self.om_doe, self.doe_logfile)
+                self.om_doe.run_driver()
+                self.om_doe.cleanup()
+            
+                # Load doe
+                doe_df = self.post_doe(save_csv=True)
+                doe_df.sort_values('r_sched.omega', inplace=True, ignore_index=True)
+
+                try:
+                    # Find initial omega
+                    om0 = np.mean(doe_df['r_sched.omega_opt'][doe_df['r_sched.sm'] > 0.0])
+                    if np.isnan(om0):
+                        raise
+                    print('Found an initial condition:', om0)
+
+                except:
+                    print('Unable to initialize om0 properly')
+                    om0 = 0.05
+
+                # Setup optimization
+                self.om_opt.set_val('r_sched.u_eval', u)
+                self.om_opt.set_val('r_sched.omega', om0)
+
+                # Run optimization
+                print('Running optimization for u = ', u)
+                opt_logfile = os.path.join(self.output_dir, self.output_name + '.' + str(u) + ".opt.sql")
+                self.om_opt = self.setup_recorder(self.om_opt, opt_logfile)
+                self.om_opt.run_driver()
+                self.om_opt.cleanup()
+
+                # save values
+                self.omegas.append(self.om_opt.get_val('r_sched.omega')[0])
+                self.k_floats.append(self.om_opt.get_val('r_sched.k_float')[0])
+                self.sms.append(self.om_opt.get_val('r_sched.sm')[0])
+
+    def plot_schedule(self):
+        fig, ax = plt.subplots(3, 1, constrained_layout=True, sharex=True)
+        ax[0].plot(self.opt_options['windspeed'], self.omegas)
+        ax[0].set_ylabel('omega_pc')
+        ax[0].grid()
+        ax[1].plot(self.opt_options['windspeed'], self.k_floats)
+        ax[1].set_ylabel('k_float')
+        ax[1].grid()
+        ax[2].plot(self.opt_options['windspeed'], self.sms)
+        ax[2].set_ylabel('stability margin')
+        ax[2].set_xlabel('Wind Speed')
+        ax[2].grid()
+        plt.show()
+
+
+    def add_dv(self, om_problem, opt_vars):
+        '''add design variables'''
+
+        if 'omega' in opt_vars and len(self.opt_options['omega']) == 2:
+            om_problem.model.add_design_var(
+                    'r_sched.omega', lower=self.opt_options['omega'][0], upper=self.opt_options['omega'][1])
+
+        if 'k_float' in opt_vars and len(self.opt_options['k_float']) == 2:
+            om_problem.model.add_design_var(
+                'r_sched.k_float', lower=self.opt_options['k_float'][0], upper=self.opt_options['k_float'][1], ref=100)
+
+        # Make sure design variables are stored appropriately in OM problem
+        om_problem.model._design_vars = om_problem.model._static_design_vars
```

Comment:
@johnjasa, I ended up doing some sort of "trickery" here to get OM to recognize the design variables when I add them after calling setting up the problem. Does this look like a reasonable way of doing this, or is there a better function to use here?